packages/wallet-protocols/src/protocols/create-and-direct-fund/protocol.ts

```diff
@@ -0,0 +1,120 @@
+import { Machine, MachineConfig, sendParent } from 'xstate';
+
+import { JsonRpcCreateChannelParams } from '../../json-rpc';
+import { MachineFactory } from '../../machine-utils';
+import { Store, success } from '../..';
+
+import { ethAllocationOutcome } from '../../calculations';
+
+import { AdvanceChannel, DirectFunding } from '..';
+
+const PROTOCOL = 'create-and-direct-fund';
+
+export enum Indices {
+  Left = 0,
+  Right = 0,
+}
+
+export type Init = JsonRpcCreateChannelParams & {
```

Comment:
Since the channel is already initialized, the `Init` type can be
```
{
  channelId: string,
  ... // whatever data is needed to construct the first state
}
```

Or, if you want to bump the responsibility of constructing the first state to the caller of this protocol, it can be
```
{
  channelId: string;
  firstState: State
}
```